clang/test/Driver/sycl-dump-device-code.cpp

```diff
@@ -0,0 +1,23 @@
+// SYCL offloading tests using -fsycl-dump-device-code
+
+// Verify that -fsycl-dump-device-code puts the device code (.spv files)
+// in the user provided directory.
+
+// clang -fsycl -target x86_64-unknown-linux-gnu
+// RUN: %clang -fsycl -fno-sycl-instrument-device-code -fno-sycl-device-lib=all -target x86_64-unknown-linux-gnu -fsycl-dump-device-code=/user/input/path/ %s -### 2>&1 \
+// RUN: | FileCheck %s --check-prefixes=CHK-FSYCL-DUMP-DEVICE-CODE
+
+// clang -fsycl -fsycl-targets=spir64-unknown-unknown
+// RUN: %clang -fsycl  -fsycl-targets=spir64-unknown-unknown -target x86_64-unknown-linux-gnu -fsycl-dump-device-code=/user/input/path/ %s -### 2>&1 \
+// RUN: | FileCheck %s --check-prefixes=CHK-FSYCL-DUMP-DEVICE-CODE
+
+// clang --driver-mode=g++
+// RUN: %clangxx -fsycl  -fsycl-targets=spir64-unknown-unknown -target x86_64-unknown-linux-gnu -fsycl-dump-device-code=/user/input/path/ %s -### 2>&1 \
```

Comment:
Use of `-fsycl-dump-device-code=/usr/input/path` should also be valid (no trailing slash).  Can you test that as well?

Reply:
Currently `-fsycl-dump-device-code=/usr/input/path` will fail since I'm invoking `llvm::sys::path::remove_filename` and we will get `/usr/input/`  as the final path.
I will remove the call to `llvm::sys::path::remove_filename` and update the test.